internal/models/payments_initiation_adjustments.go

```diff
@@ -0,0 +1,74 @@
+package models
+
+import (
+	"encoding/json"
+	"time"
+)
+
+type PaymentInitiationAdjustment struct {
+	// Unique ID
+	ID PaymentInitiationAdjustmentID `json:"id"`
+
+	// Related Payment Initiation ID
+	PaymentInitiationID PaymentInitiationID `json:"paymentInitiationID"`
+	// Creation date of the adjustment
+	CreatedAt time.Time `json:"createdAt"`
+	// Last status of the adjustment
+	Status PaymentInitiationAdjustmentStatus `json:"status"`
+	// Error description if we had one
+	Error *string `json:"error"`
```

Comment:
```suggestion
	Error *string `json:"error,omitempty"`
```

Since you have a custom unmarshal function, you could probably also use type `error` in the public struct

Reply:
You're right, I put the type error everywhere instead of *string, it's cleaner !